app/models/page.rb

```diff
@@ -36,7 +36,8 @@ def body_present
   end
 
   def unique_slug_including_users_and_orgs
-    errors.add(:slug, "is taken.") if User.find_by(username: slug) || Organization.find_by(slug: slug) || Podcast.find_by(slug: slug)
+    slug_exists = User.exists?(username: slug) || Organization.exists?(slug: slug) || Podcast.exists?(slug: slug)
```

Comment:
here I replaced `find_by` which loads the entire object in memory, with `exists?` which only returns a boolean